src/etl/normalizer.py

```diff
@@ -0,0 +1,223 @@
+import re
+import polars as pl
+import pycountry
+from logging import Logger
+
+from utilities.units_converter import converter_map
+
+
+class Normalizer:
+    def __init__(self, config: dict, logger: Logger) -> None:
+        self.logger = logger
+        self.data_config = config["data_config"]
+        self.data_columns = config["data_columns"]
+        self.data_normalizer = config["data_normalizer"]
+
+    def execute(self, df_base: pl.DataFrame) -> pl.DataFrame:
+        """
+        Execute all the methods defined in the normalizer.
+        """
+        pipeline = [
+            self.normalize_inches_columns,
+            self.normalize_pound_columns,
+            self.convert_units_to_float,
+            self.convert_currency,
+            self.filter_wrong_upcs,
+            self.convert_upc_to_ean13,
+            self.rename_columns,
+            self.normalize_country_column,
+            self.format_date_columns,
+            self.convert_int_columns,
+        ]
+
+        for step in pipeline:
+            df_base = step(df_base)
+
+        return df_base
+
+    def convert_units(
+        self,
+        df_base: pl.DataFrame,
+        valid_units: str,
+        pattern: re.Pattern,
+        replacer: str,
+    ) -> pl.DataFrame:
+        """
+        Function to convert the units.
+
+        Parameters
+        ----------
+        df_base: pl.DataFrame
+            The input DataFrame.
+        valid_units: str
+            The valid units to convert.
+        pattern: re.Pattern
+            The regular expression to search for the unit in the column.
+        replacer: str
+            The replacement string for the new column name.
+
+        Returns
+        -------
+            pl.DataFrame
+        """
+
+        # Get all the columns with valids units.
+        if columns_with_units := df_base.select(
+            rf"^.*\(({valid_units})\).*$"
+        ).columns:
+            # For each column, it will convert the values according
+            # to the unit matched by the regex.
+            for col in columns_with_units:
+                result = pattern.findall(col)[0]
+                # Replace the old unit with the corresponding one. (inches or pounds)
+                new_name = pattern.sub(replacer, col)
+                df_base = df_base.with_columns(
+                    pl.col(col)
+                    .cast(pl.Float64)
+                    .map_elements(converter_map[result])
+                    .alias(new_name)
+                ).drop(col)
+
+        return df_base
+
+    def normalize_inches_columns(self, df_base: pl.DataFrame) -> pl.DataFrame:
+        """
+        Converts column values that may come in units other than inches.
+        It converts from centimeters, meter and feet to inches.
+        """
+        valid_units = "|".join(self.data_config["inches_units"])
```

Comment:
Call `map(re.escape, ...)` to safely join iterables of strings to be interpolated into regex.